code/style.css

```diff
@@ -1,4 +1,367 @@
-html {
-  background: #0025ff;
-  color: #fff;
+body {
+    font-family: 'Inconsolata', monospace;
+    font-family: 'Mukta Vaani', sans-serif;
+    margin: 0;
+    padding: 0;
 }
+
+.main-grid {
+    display: grid;
+    grid-template-columns: repeat(12, 1fr);
+}
+
+.header {
+    grid-column: span 12;
+    color: black;
+    padding: 5px;
+    text-transform: uppercase;
+    font-weight: 300;
+    letter-spacing: 3px;
+}
+
+.header ul {
+    margin: 0;
+    padding: 0;
+    list-style: none;
+    overflow: hidden;
+    background-color: #fff;
+}
+
+.header li a {
+    display: block;
+    padding: 10px 20px;
+    text-decoration: none;
+}
+
+/* nav start */
+
+.header a {
+    color: black;
+}
+
+.menu a:hover {
+    color: red;
+}
+
+.header .menu {
+    clear: both;
+    max-height: 0;
+    transition: max-height .2s ease-out;
+}
+
+.header .logo {
+    align-items: center;
+    display: block;
+    float: left;
+    font-size: 24px;
+    padding: 10px 3px 3px 3px;
+    text-decoration: none;
+}
+
+.header .menu-icon {
+    cursor: pointer;
+    float: right;
+    padding: 28px 20px;
+    position: relative;
+    user-select: none;
+}
+
+.header .menu-icon .navicon {
+    background: #333;
+    display: block;
+    height: 2px;
+    position: relative;
+    transition: background .2s ease-out;
+    width: 18px;
+}
+
+.header .menu-icon .navicon:before, .header .menu-icon .navicon:after {
+    background: #333;
+    content: '';
+    display: block;
+    height: 100%;
+    position: absolute;
+    transition: all .2s ease-out;
+    width: 100%;
+}
+
+.header .menu-icon .navicon:before {
+    top: 5px;
+}
+
+.header .menu-icon .navicon:after {
+    top: -5px;
+}
+
+.header .menu-btn {
+    display: none;
+}
+
+.header .menu-btn:checked~.menu {
+    max-height: 200px;
+}
+
+.header .menu-btn:checked~.menu-icon .navicon {
+    background: transparent;
+}
+
+.header .menu-btn:checked~.menu-icon .navicon:before {
+    transform: rotate(-45deg);
+}
+
+.header .menu-btn:checked~.menu-icon .navicon:after {
+    transform: rotate(45deg);
+}
+
+.header .menu-btn:checked~.menu-icon:not(.steps) .navicon:before, .header .menu-btn:checked~.menu-icon:not(.steps) .navicon:after {
```

Comment:
**Description:**
It’s the icon and menu animations when the checkbox is clicked. So from line 103 to 115 you see all the stages in the animation. The last line 115 describes all the selected elements that is preceded by the former selector.


**The (tilde) ~ sign** 
It is general sibling combinator and similar to Adjacent sibling combinator. the difference is that the second selector does NOT have to immediately follow the first one means It will select all elements that is preceded by the former selector.